src/protocol-handler-installer.js

```diff
@@ -0,0 +1,101 @@
+const {CompositeDisposable} = require('event-kit')
+
+const {remote} = require('electron')
+
+function isSupported () {
+  return ['win32', 'darwin'].includes(process.platform)
+}
+
+function isDefaultProtocolClient () {
+  return remote.app.isDefaultProtocolClient('atom', process.execPath, ['--url-handler'])
+}
+
+function setAsDefaultProtocolClient () {
+  // This Electron API is only available on Windows and macOS. There might be some
+  // hacks to make it work on Linux; see https://github.com/electron/electron/issues/6440
+  return isSupported() && remote.app.setAsDefaultProtocolClient('atom', process.execPath, ['--url-handler'])
+}
+
+module.exports =
+class ProtocolHandlerInstaller {
+  constructor () {
+    this.subscriptions = new CompositeDisposable()
+  }
+
+  initialize (config, notifications) {
+    this.config = config
+    this.notifications = notifications
+
+    this.subscriptions.add(this.config.observe('core.uriHandlerRegistration', this.onValueChange.bind(this)))
```

Comment:
🤔  I wonder if it's too obnoxious to immediately prompt the user if they switch to `"prompt"` and Atom isn't the default protocol handler — maybe we should only check at startup?